.github/workflows/build-check.yaml

```diff
@@ -0,0 +1,33 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        arch: [x86_64, aarch64, loongarch64, powerpc64le, wasm]
```

Comment:
I would probably skip loongarch and powerpc, because they are not that common.

Reply:
I added them because the motivation to create it came from this PRhttps://github.com/storj/uplink-c/pull/30 which has it as a supported platform to cross compile https://github.com/storj/uplink-c/pull/30

Reply:
ok, let's leave them in.